### PR TITLE
[runtime] Use length of stacktrace rather than SP for bitcode watermark

### PR DESCRIPTION
**Pull request by alexanderkyte**

### CODE REVIEW

File: mono/mini/mini-exceptions.c [hunk 1]
```diff
@@ -128,29 +130,49 @@ mono_thread_get_managed_sp (void)
...
+	}
+}
 
-	return direction * magnitude;
+static inline size_t
+mini_thread_get_threshold_scalar (void)
+{
+	if (mono_llvm_only) {
+	  size_t count = count_unwind_backtrace ();
+		return count;
```
Comment (rjvdboon) [inline mono/mini/mini-exceptions.c]: nit (sorry): spacing.

File: mono/mini/mini-exceptions.c [hunk 2]
```diff
@@ -128,29 +130,49 @@ mono_thread_get_managed_sp (void)
 }
 
 static inline int
-mini_abort_threshold_offset (gpointer threshold, gpointer sp)
+mini_abort_threshold_offset (size_t stack_threshold, size_t stack_size)
 {
-	intptr_t stack_threshold = (intptr_t) threshold;
```
Comment (vargaz) [inline mono/mini/mini-exceptions.c]: Can this get a comment saying that the arguments are sometimes sp, and sometimes stack size ?

File: mono/mini/mini-exceptions.c (hunk 1 above)
Comment (alexanderkyte) [inline mono/mini/mini-exceptions.c]: Thanks! I've fixed it.

File: mono/mini/mini-exceptions.c (hunk 2 above)
Comment (alexanderkyte) [inline mono/mini/mini-exceptions.c]: It's been added.

**Pull request closed without merging**